docs/source/posts/2024/2024-07-06-week5-wachiou-bouraima.rst

```diff
@@ -0,0 +1,112 @@
+WEEK 5: Implementing Lazy Loading in FURY with ``lazy_loader``
+==============================================================
+
+.. post:: July 6, 2024
+   :author: Wachiou BOURAIMA
+   :tags: google
+   :category: gsoc
+
+Hello everyone,
+---------------
+
+Welcome back to my Google Summer of Code (GSoC) 2024 journey! This week has been particularly exciting as I introduced a significant performance optimization feature: lazy loading. Here's an overview of my progress and contributions.
+
+
+**Introduction of lazy loading**
+--------------------------------
+
+This week, I focused on implementing the ``lazy_loader`` feature of `Scientific Python <https://scientific-python.org/>`_ to optimize module loading in FURY. Lazy loading improves performance by deferring the loading of modules until they are actually needed, thus reducing start-up times and memory footprint.
+
+The implementation involved:
+
+1. Implementation of Lazy Loading:
+
+   - Application of lazy loading in several FURY modules using the ``lazy_loader`` module to improve performance
+
+2. Update ``__init__.py`` files:
+
+   - Modified ``__init__.py`` files to support lazy loading where necessary. This ensures that modules are only loaded when they are accessed for the first time
+
+3. Added Type Stubs (``__init__.pyi``):
+
+   - Adding type stubs (``__init__.pyi``) provides type hints for lazy-loading modules, improving code readability and maintainability
+
+4. **Improved module organization:**
+
+   - Improved module organization and structure to effectively support lazy loading.
```

Comment:
How did you do this? I mean, what is the rationale behind improving the structure?

Reply:
Merci @itellaetxe for your feedback. 

I made a first implementation of lazy loading in which I imported modules and functions into __init__.pyi files like this:
``
from .actor import (
    Container,
    _color_fa,
    ...
)
``

then I restified and improved it like this

``from .actor import (
    Container as Container,
    _color_fa as _color_fa,
    ...
)
``
to comply with SPEC1 recommendations: https://scientific-python.org/specs/spec-0001/#type-checkers